src/lesson-3/modules/tasks/task.service.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  Bug,
+  Epic,
+  FilterParams,
+  NewTaskBody,
+  Story,
+  Subtask,
+  TaskDetails,
+  TaskType,
+  UnitedSchema,
+} from './task.types';
+
+export class TaskService {
+  #tasks: UnitedSchema[];
+
+  constructor(tasks: UnitedSchema[]) {
+    this.#tasks = tasks;
+  }
+
+  getAll() {
+    return this.#tasks;
+  }
+
+  // create new Task
+  createNewInstance({
+    type,
+    taskDetail,
+  }: {
+    type: TaskType;
+    taskDetail: NewTaskBody;
+  }) {
+    let task!: UnitedSchema;
+
+    if (type === 'bug') {
+      task = new Bug({ ...taskDetail, createdAt: new Date() });
```

Comment:
можна було б винести створюваний обʼєкт у окрему змінну, щоб не дублювати код
```suggestion
      const taskData = { ...taskDetail, createdAt: new Date() }
      ....
      task = new Bug(taskData);
```